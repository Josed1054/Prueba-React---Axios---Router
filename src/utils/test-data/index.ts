export const Data = [
  {
    _id: "1",
    cityid: "1",
    name: "Uno",
    probabilityofprecip: 60,
    relativehumidity: 20,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "2",
    cityid: "2",
    name: "Dos",
    probabilityofprecip: 71,
    relativehumidity: 52,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "3",
    cityid: "3",
    name: "Tres",
    probabilityofprecip: 60,
    relativehumidity: 22,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "4",
    cityid: "4",
    name: "Cuatro",
    probabilityofprecip: 65,
    relativehumidity: 20,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "5",
    cityid: "5",
    name: "Cinco",
    probabilityofprecip: 60,
    relativehumidity: 20,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "6",
    cityid: "6",
    name: "Seis",
    probabilityofprecip: 60,
    relativehumidity: 70,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "7",
    cityid: "7",
    name: "Siete",
    probabilityofprecip: 80,
    relativehumidity: 90,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "8",
    cityid: "8",
    name: "Ocho",
    probabilityofprecip: 30,
    relativehumidity: 20,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "9",
    cityid: "9",
    name: "Nueve",
    probabilityofprecip: 60,
    relativehumidity: 60,
    lastreporttime: "20170627T092449Z",
  },
  {
    _id: "10",
    cityid: "10",
    name: "Diez",
    probabilityofprecip: 70,
    relativehumidity: 40,
    lastreporttime: "20170627T092449Z",
  },
];
