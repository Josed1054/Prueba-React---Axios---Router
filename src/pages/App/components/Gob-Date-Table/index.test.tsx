import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { GobDataTable } from ".";
import { MemoryRouter as Router } from "react-router-dom";
import { Data } from "../../../../utils/test-data/index";

test("Checar si se render el pedaso de tabla", () => {
  const view = renderer
    .create(
      <Router>
        <table>
          <tbody>
            <GobDataTable gobData={Data} numberOfEntries={10} />
          </tbody>
        </table>
      </Router>
    )
    .toJSON();

  expect(view).toMatchSnapshot();
});

test("Checar que no se render el pedaso de tabla", () => {
  const view = renderer
    .create(
      <Router>
        <table>
          <tbody>
            <GobDataTable gobData={Data} numberOfEntries={20} />
          </tbody>
        </table>
      </Router>
    )
    .toJSON();

  expect(view).toMatchSnapshot();
});
