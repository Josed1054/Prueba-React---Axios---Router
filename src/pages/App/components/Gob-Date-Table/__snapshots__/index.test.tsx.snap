// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checar que no se render el pedaso de tabla 1`] = `
<table>
  <tbody />
</table>
`;

exports[`Checar si se render el pedaso de tabla 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <a
          href="/1"
          onClick={[Function]}
        >
          1
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Uno
      </td>
      <td />
      <td>
        60
      </td>
      <td>
        20
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        No
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <a
          href="/2"
          onClick={[Function]}
        >
          2
        </a>
      </td>
      <td>
        2
      </td>
      <td>
        Dos
      </td>
      <td />
      <td>
        71
      </td>
      <td>
        52
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        <a
          href="/3"
          onClick={[Function]}
        >
          3
        </a>
      </td>
      <td>
        3
      </td>
      <td>
        Tres
      </td>
      <td />
      <td>
        60
      </td>
      <td>
        22
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        No
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        <a
          href="/4"
          onClick={[Function]}
        >
          4
        </a>
      </td>
      <td>
        4
      </td>
      <td>
        Cuatro
      </td>
      <td />
      <td>
        65
      </td>
      <td>
        20
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        <a
          href="/5"
          onClick={[Function]}
        >
          5
        </a>
      </td>
      <td>
        5
      </td>
      <td>
        Cinco
      </td>
      <td />
      <td>
        60
      </td>
      <td>
        20
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        No
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        <a
          href="/6"
          onClick={[Function]}
        >
          6
        </a>
      </td>
      <td>
        6
      </td>
      <td>
        Seis
      </td>
      <td />
      <td>
        60
      </td>
      <td>
        70
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        <a
          href="/7"
          onClick={[Function]}
        >
          7
        </a>
      </td>
      <td>
        7
      </td>
      <td>
        Siete
      </td>
      <td />
      <td>
        80
      </td>
      <td>
        90
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
    <tr>
      <td>
        8
      </td>
      <td>
        <a
          href="/8"
          onClick={[Function]}
        >
          8
        </a>
      </td>
      <td>
        8
      </td>
      <td>
        Ocho
      </td>
      <td />
      <td>
        30
      </td>
      <td>
        20
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        No
      </td>
    </tr>
    <tr>
      <td>
        9
      </td>
      <td>
        <a
          href="/9"
          onClick={[Function]}
        >
          9
        </a>
      </td>
      <td>
        9
      </td>
      <td>
        Nueve
      </td>
      <td />
      <td>
        60
      </td>
      <td>
        60
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
    <tr>
      <td>
        10
      </td>
      <td>
        <a
          href="/10"
          onClick={[Function]}
        >
          10
        </a>
      </td>
      <td>
        10
      </td>
      <td>
        Diez
      </td>
      <td />
      <td>
        70
      </td>
      <td>
        40
      </td>
      <td>
        2017/06/27
      </td>
      <td>
        Si
      </td>
    </tr>
  </tbody>
</table>
`;
